### PR TITLE
Add `Workspace.registered_run_names()` method

### PR DESCRIPTION
**Pull request by epwalsh**
Alternative to #489 as we discussed this morning @dirkgr

### CODE REVIEW

File: tango/workspace.py [hunk 1]
```diff
@@ -234,6 +234,17 @@ def registered_runs(self) -> Dict[str, Run]:
         """
         raise NotImplementedError()
 
+    def registered_run_names(self) -> List[str]:
+        """
+        Returns a list of all runs in the workspace, ideally sorted by creation time, newest first.
```
[the remark this answers is not in the document]
Comment (dirkgr) [reply tango/workspace.py:239]: As in, we lose that ability even in the UI?

[the remark this answers is not in the document]
Comment (epwalsh) [reply tango/workspace.py:239]: Yea, we can filter/sort by name at least, and potentially sort by start time but only if we enforce that ordering in the API for this method.

[the remark this answers is not in the document]
Comment (dirkgr) [reply tango/workspace.py:239]: We're going to have the same problem again for steps, right?

[the remark this answers is not in the document]
Comment (epwalsh) [reply tango/workspace.py:239]: Yea good point. I guess we also need a `.registered_step_names()` method or something.

**Pull request closed without merging**